Vostok.Metrics.System/Host/DiskUsageCollector_Linux.cs

```diff
@@ -0,0 +1,115 @@
+using System.Diagnostics;
+using System.Linq;
+using Vostok.Metrics.System.Helpers;
+
+namespace Vostok.Metrics.System.Host
+{
+    internal class DiskUsageCollector_Linux
+    {
+        private Dictionary<string, DiskStats> previousDisksStatsInfo = new Dictionary<string, DiskStats>();
+        private readonly Stopwatch stopwatch = new Stopwatch();
+
+        public void Collect(HostMetrics metrics, IEnumerable<string> diskstatsLines)
+        {
+            var deltaTime = (double) stopwatch.ElapsedTicks / Stopwatch.Frequency;
+
+            var disksUsageInfo = new Dictionary<string, DiskUsageInfo>();
+
+            foreach (var diskStats in ParseDiskstats(diskstatsLines))
+            {
+                var result = new DiskUsageInfo {DiskName = diskStats.DiskName};
+
+                if (previousDisksStatsInfo.ContainsKey(diskStats.DiskName))
+                    FillInfo(result, previousDisksStatsInfo[diskStats.DiskName], diskStats, deltaTime);
+
+                disksUsageInfo[result.DiskName] = result;
+
+                previousDisksStatsInfo[diskStats.DiskName] = diskStats;
+            }
+
+            metrics.DisksUsageInfo = disksUsageInfo;
+
+            RemoveUnusedEntries(disksUsageInfo);
+
+            stopwatch.Restart();
+        }
+
+        private void FillInfo(DiskUsageInfo toFill, DiskStats previousDiskStats, DiskStats diskStats, double deltaTime)
+        {
+            var readsDelta = diskStats.ReadsCount - previousDiskStats.ReadsCount;
+            var writesDelta = diskStats.WritesCount - previousDiskStats.WritesCount;
+            var timeReadDelta = diskStats.TimeSpentReading - diskStats.TimeSpentReading;
+            var timeWriteDelta = diskStats.TimeSpentWriting - diskStats.TimeSpentWriting;
+
+            if (readsDelta > 0)
+                toFill.ReadLatency = (double) timeReadDelta / readsDelta;
+            if (writesDelta > 0)
+                toFill.WriteLatency = (double) timeWriteDelta / writesDelta;
+
+            toFill.CurrentQueueLength = diskStats.CurrentQueueLength - previousDiskStats.CurrentQueueLength;
+
+            toFill.ReadsPerSecond = readsDelta / deltaTime;
+            toFill.WritesPerSecond = writesDelta / deltaTime;
+
+            toFill.IdleTimePercent = ((1 - (timeReadDelta + timeWriteDelta) / deltaTime) * 100).Clamp(0, 100);
+        }
+
+        private void RemoveUnusedEntries(Dictionary<string, DiskUsageInfo> disksUsageInfo)
+        {
+            previousDisksStatsInfo = previousDisksStatsInfo
+               .Where(x => disksUsageInfo.ContainsKey(x.Key))
+               .ToDictionary(x => x.Key, y => y.Value);
+        }
+
+        private List<DiskStats> ParseDiskstats(IEnumerable<string> diskstats)
+        {
+            var disksStats = new List<DiskStats>();
+
+            try
+            {
+                foreach (var line in diskstats)
+                {
+                    if (FileParser.TrySplitLine(line, 14, out var parts) && parts[1] == "0")
```

Comment:
1. Why is it required to have 14 components in the line? You only seem to use up to 12, and [documentation](https://www.kernel.org/doc/Documentation/ABI/testing/procfs-diskstats) lists 13 in the most basic form.

Reply:
2. `parts[1] == "0"` is a rather esoteric, "magic" piece of code. It's not clear what it does and why it exists. Please provide a valid explanation for such choice (preferably in the code itself, a detailed comment would suffice).